helm/templates/daemonset.yaml

```diff
@@ -3,7 +3,9 @@ kind: DaemonSet
 apiVersion: apps/v1
 metadata:
   name: efs-csi-node
-  namespace: kube-system
+  namespace: {{ .Values.namespace }}
```

Comment:
I am not sure about this project but in general you either hard code a namespace because it is required to run in a specific namespace or you leave it out completely and allow the user to decide which namespace is goes in based on the namespace flag with the helm command.  I would generally not expect to have to set a value to control the namespace.  Maybe an owner would weigh in on whether to leave it hardcoded or remove it in favor of typical helm behavior.

@d-nishi @leakingtapan @justinsb @jsafrane @wongma7 @nckturner @jqmichael

Reply:
I had trouble with this, originally I had removed the hardcoded namespace because I felt that it would be up to the individual who installed the driver where they would like it to do. The problem is that once I was OK to test, the tests assume the chart is installed in kube-system, so without a default namespace set, it installs in the default namespace and the tests fail. Wasn't sure what to do so I went with a middle approach.

Reply:
I haven't looked at these tests much but I think at least this helm install would need to change to have `--namespace kube-system` added to it if the hard coded namespace were removed.
https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/eed29e4c149e50452e8f030f31ee05736b1fa160/tester/e2e-test-config.yaml#L30

Reply:
Thanks for pointing that out, I changed the install to include the namespace and went back to my original commit with using the namespace from the release. e2e works perfectly now and this is the behavior that I would expect.

Reply:
it seems subject to a lot of debate. https://github.com/helm/helm/issues/5465 I don't mind whether we hardcode the namespace or do this Release.namespace thing, as long as the `helm install` in the test script AND the README work.